ObjectiveGit/GTRepository+Merging.m

```diff
@@ -170,6 +172,87 @@ - (BOOL)mergeBranchIntoCurrentBranch:(GTBranch *)branch withError:(NSError **)er
 	return NO;
 }
 
+- (NSString * _Nullable)contentsOfDiffWithAncestor:(GTIndexEntry *)ancestor ourSide:(GTIndexEntry *)ourSide theirSide:(GTIndexEntry *)theirSide error:(NSError **)error {
+
+	GTObjectDatabase *database = [self objectDatabaseWithError:error];
+	if (database == nil) {
+		return nil;
+	}
+
+	// initialize the ancestor's merge file input
+	git_merge_file_input ancestorInput;
+	int gitError = git_merge_file_init_input(&ancestorInput, GIT_MERGE_FILE_INPUT_VERSION);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create merge file input for ancestor"];
+		return nil;
+	}
+
+	git_oid ancestorId = ancestor.git_index_entry->id;
+	GTOID *ancestorOID = [[GTOID alloc] initWithGitOid:&ancestorId];
+	NSData *ancestorData = [[database objectWithOID:ancestorOID error: error] data];
+	if (ancestorData == nil) {
+		return nil;
+	}
+	ancestorInput.ptr = ancestorData.bytes;
+	ancestorInput.size = ancestorData.length;
+
+
+	// initialize our merge file input
+	git_merge_file_input ourInput;
+	gitError = git_merge_file_init_input(&ourInput, GIT_MERGE_FILE_INPUT_VERSION);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create merge file input for our side"];
+		return nil;
+	}
+
+	git_oid ourId = ourSide.git_index_entry->id;
+	GTOID *ourOID = [[GTOID alloc] initWithGitOid:&ourId];
+	NSData *ourData = [[database objectWithOID:ourOID error: error] data];
+	if (ourData == nil) {
+		return nil;
+	}
+	ourInput.ptr = ourData.bytes;
+	ourInput.size = ourData.length;
+
+
+	// initialize their merge file input
+	git_merge_file_input theirInput;
+	gitError = git_merge_file_init_input(&theirInput, GIT_MERGE_FILE_INPUT_VERSION);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create merge file input other side"];
+		return nil;
+	}
+
+	git_oid theirId = theirSide.git_index_entry->id;
+	GTOID *theirOID = [[GTOID alloc] initWithGitOid:&theirId];
+	NSData *theirData = [[database objectWithOID:theirOID error: error] data];
+	if (theirData == nil) {
+		return nil;
+	}
+	theirInput.ptr = theirData.bytes;
+	theirInput.size = theirData.length;
+
+
+	git_merge_file_result result;
+	gitError = git_merge_file(&result, &ancestorInput, &ourInput, &theirInput, nil);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create merge file"];
+		return nil;
+	}
+
+	char *cString = malloc(result.len * sizeof(char *) + 1);
+	strncpy(cString, result.ptr, result.len);
+	cString[result.len] = '\0';
+
+	NSString *mergedContent = [[NSString alloc] initWithCString:cString encoding:NSUTF8StringEncoding];
```

Comment:
Maybe `-initWithBytesNoCopy:length:encoding:freeWhenDone:` ? That should allow you to just pass `result.ptr`/`result.len`, skipping the need for the `malloc`/`strncpy` step. You would have to `result.ptr = NULL` though, so `git_merge_file_result_free` doesn't break it by freeing before returning.